jobbergateapi2/tests/apps/application_permissions/test_routers.py

```diff
@@ -0,0 +1,247 @@
+    assert count[0][0] == 0
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_list_application_permissions(user_data, client):
+    """
+    Test the GET in the /application-permissions/ returns the list of existing ApplicationPermissions.
+
+    We show this by creating 2 permissions, making the GET request to the /application-permissions/ endpoint,
+    then asserting the response status code is 200, and the content of the response is as expected.
+    """
+    user = [UserCreate(**user_data)]
+    await insert_objects(user, users_table)
+
+    application_permissions = [
+        ApplicationPermission(id=1, acl="Allow|role:admin|create"),
+        ApplicationPermission(id=2, acl="Allow|role:admin|view"),
+    ]
+    await insert_objects(application_permissions, application_permissions_table)
+
+    response = client.get("/application-permissions/")
+    assert response.status_code == status.HTTP_200_OK
+
+    data = response.json()
+    assert len(data) == 2
+
+    assert data[0]["id"] == 1
+    assert data[0]["acl"] == "Allow|role:admin|create"
+    assert data[1]["id"] == 2
+    assert data[1]["acl"] == "Allow|role:admin|view"
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_list_application_permissions_empty(user_data, client):
+    """
+    Test the GET in the /application-permissions/ returns the list of existing ApplicationPermissions even
+    when there is none.
+
+    We show this by making a request to the /application-permissions/ endpoint, asserting that the response
+    status code is 200 and that the contents of the response is empty.
+    """
+    user = [UserCreate(**user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.get("/application-permissions/")
+    assert response.status_code == status.HTTP_200_OK
+
+    data = response.json()
+    assert len(data) == 0
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_delete_application_permissions(user_data, client):
+    """
+    Test DELETE /application-permissions/{id} by a superuser correctly deletes the ApplicationPermission.
+
+    This test proves that an ApplicationPermission is successfully deleted via a DELETE request to the
+    /application-permissions/. We show this this asserting that the ApplicationPermission no longer exists
+    in the database and the correct status code (204) is returned.
+    """
+    user = [UserCreate(is_superuser=True, **user_data)]
+    await insert_objects(user, users_table)
+
+    application_permissions = [
+        ApplicationPermission(id=1, acl="Allow|role:admin|create"),
+        ApplicationPermission(id=2, acl="Allow|role:admin|view"),
+    ]
+    await insert_objects(application_permissions, application_permissions_table)
+
+    response = client.delete("/application-permissions/1")
+    assert response.status_code == status.HTTP_204_NO_CONTENT
```

Comment:
I believe this should return a 200 success if we are trying to delete 1, and then next we check that 1 is deleted and 2 remains.

Reply:
About the status code, here we return 204  since there is no body in the response, 200 would fit if we were expecting something if the response body, which is not the case.